### PR TITLE
Add Pipeline with Dockerfile Environment

### PR DESCRIPTION
**Pull request by kottofy**
New PR due to issues with running Workflows with branches in forked repos.

- Adds a sample taxi pipeline with a custom Docker environment
- Allows for AML environment to be created with either conda or Docker environment
- Adds GitHub workflows and Azure Pipelines to run unit tests with docker

### CODE REVIEW

File: .azure-pipelines/templates/build_validation_pipeline.yml [hunk 1]
```diff
@@ -1,5 +1,53 @@
...
+  steps:
+  - script: |
+      python -m pip install --upgrade pip
+      pip install -r .azure-pipelines/requirements/build_validation_requirements.txt
+      flake8 .
+    displayName: 'Install dependencies and run linting'
+
+  - script: | 
+      echo "Building Docker container for validation with tag: ${{ parameters.image_name }}"
+      docker build -t ${{ parameters.image_name }} -f mlops/${{ parameters.model_type }}/environment/Dockerfile mlops/${{ parameters.model_type }}/environment
```
Comment (agonzalez2014) [inline .azure-pipelines/templates/build_validation_pipeline.yml:25]: Please find time to have a design conversation with @sbaidachni on this approach for enabling container images to be used for CI and CD operations. Worthwhile to think if this is acceptable to go in as is (with any GH issues for optimizations/enhancements), or if we want to invest time in considering alternatives.

File: .github/workflows/register_data_assets.yml [hunk 2]
```diff
@@ -1,7 +1,12 @@
 name: Register Datasets
 
+# TODO restore
+# on:
+#   workflow_dispatch:
```
Comment (agonzalez2014) [inline .github/workflows/register_data_assets.yml:5]: is the intent to execute this on PR or to be triggered automatically? is the TOOD / commented out code needed?

Comment (kottofy) [reply .github/workflows/register_data_assets.yml:5]: This workflow is ran manually. I need it to register the new data assets for the docker_taxi pipeline before the other new workflows can work. I will restore and remove this TODO once the workflows are past the storage account authentication issue. Let's leave this comment up to help me remember to do that :)

File: README.md [hunk 3]
```diff
@@ -6,26 +6,25 @@ The idea of this template is to provide a minimum number of scripts to implement
...
-- mlops: scripts that are related to Azure ML.
-- mlops/london-taxi, mlops/nyc-taxi: fake pipelines with some basic code.
-- .amlignore: using this file we are removing all the folders and files that are not supposed to be in Azure ML compute.
+- [.github](.github): the folder contains Github Workflow related files (yaml files to define the Builds).
+- [.azure-pipelines](.azure-pipelines): the folder contains Azure DevOps related files (yaml files to define Builds).
+- [docs](docs): documentation.
+- [src](src): source code that is not related to Azure ML directly. Usually, there is data science related code.
+- [mlops](mlops): scripts that are related to Azure ML.
+- [mlops/london-taxi](mlops/london-taxi), [mlops/nyc-taxi](mlops/nyc-taxi), [mlops/docker_taxi](mlops/docker_taxi): fake pipelines with some basic code.
+- [.amlignore](.amlignore): using this file we are removing all the folders and files that are not supposed to be in Azure ML compute.
```
Comment (agonzalez2014) [inline README.md:15]: love this :)

File: .github/workflows/register_data_assets.yml (hunk 2 above)
Comment (kottofy) [reply .github/workflows/register_data_assets.yml]: All restored!

**Review by agonzalez2014 — APPROVED** [review]

**Review by marshallbentley — APPROVED** [review]

File: .azure-pipelines/templates/build_validation_pipeline.yml (hunk 1 above)
Comment (kottofy) [reply .azure-pipelines/templates/build_validation_pipeline.yml:25]: Resolving for now as this has approvals to be merged in. Will make a new GH issue to discuss this.

**Pull request merged**